bundles/io/org.eclipse.smarthome.io.transport.serial.ctrl/src/main/java/org/eclipse/smarthome/io/transport/serial/internal/SerialPortRegistry.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.transport.serial.internal;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
+
+import org.eclipse.smarthome.io.transport.serial.ProtocolType.PathType;
+import org.eclipse.smarthome.io.transport.serial.SerialPortProvider;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author MatthiasS
+ *
+ */
+@Component(immediate = true, service = SerialPortRegistry.class)
+public class SerialPortRegistry {
+
+    private final static Logger logger = LoggerFactory.getLogger(SerialPortRegistry.class);
+
+    private Collection<SerialPortProvider> portCreators;
+
+    public SerialPortRegistry() {
+        this.portCreators = new HashSet<>();
+    }
+
+    /**
+     * Registers a {@link SerialPortProvider}.
+     *
+     * @param creator
+     */
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC, policyOption = ReferencePolicyOption.GREEDY)
+    protected void registerSerialPortCreator(SerialPortProvider creator) {
+        this.portCreators.add(creator);
```

Comment:
As you are using a dynamic reference the member portCreators will be accessed by different threads.
You could e.g. use a concurrent collection or other synchronization statements.